java/server/src/org/openqa/grid/internal/RemoteProxy.java

```diff
@@ -39,6 +39,16 @@
  * only support Firefox.
  */
 public interface RemoteProxy extends Comparable<RemoteProxy> {
+
+  /**
+   *
+   * @param protocol - A {@link SeleniumProtocol} object that identifies the request flavor.
+   * @param capabilities - the type of test the client is interested in performing.
+   * @return - The entity on a proxy that can host a test session.
+   */
+  default TestSlot newTestSlot(SeleniumProtocol protocol, Map<String, Object> capabilities) {
```

Comment:
@mach6 - Yes I very much can. But I would like to understand the intent of the functional test. It can at the max just do an `instanceof` check in the assertion. So can you please help me understand what would the test actually be doing as a value add. There is no change in code. The only thing that has been done in this PR is to move the instantiation of the `TestSlot` to an interface (which has been given a default behaviour as well)